src/controllers/crate_owner_invitation.rs

```diff
@@ -83,6 +55,200 @@ pub fn list(req: &mut dyn RequestExt) -> EndpointResult {
     }))
 }
 
+/// Handles the `GET /api/private/crate-owner-invitations` route.
+pub fn private_list(req: &mut dyn RequestExt) -> EndpointResult {
+    let auth = req.authenticate()?.forbid_api_token_auth()?;
+
+    let filter = if let Some(crate_name) = req.query().get("crate_name") {
+        ListFilter::CrateName(crate_name.clone())
+    } else if let Some(id) = req.query().get("invitee_id").and_then(|i| i.parse().ok()) {
+        ListFilter::InviteeId(id)
+    } else {
+        return Err(bad_request("missing or invalid filter"));
+    };
+
+    let list = prepare_list(req, auth, filter)?;
+    Ok(req.json(&list))
+}
+
+enum ListFilter {
+    CrateName(String),
+    InviteeId(i32),
+}
+
+fn prepare_list(
+    req: &mut dyn RequestExt,
+    auth: AuthenticatedUser,
+    filter: ListFilter,
+) -> AppResult<PrivateListResponse> {
+    let pagination: PaginationOptions = PaginationOptions::builder()
+        .enable_pages(false)
+        .enable_seek(true)
+        .gather(req)?;
```

Comment:
if this function is unconditionally using seek pagination then isn't this a breaking change for the existing endpoint? 🤔

Reply:
Yes, but as I mentioned in the PR body I'd like for the existing endpoint to be removed once we switch the frontend to using the new endpoint. The existing endpoint also rejects token requests, so the only consumer of it is the frontend.

There also shouldn't be any user visibly affected by this change: when writing this only 14 users have pending unexpired invitations, and the most invites an user has is 11:

```
crates-io::DATABASE=> select count(*) from crate_owner_invitations where created_at > current_timestamp - interval '30 days' group by invited_user_id order by count(*) desc;
 count 
-------
    11
     4
     2
     1
     1
     1
     1
     1
     1
     1
     1
     1
     1
     1
(14 rows)
```